### DISCUSSION TITLE
使用Texstudio无法正确编译参考文献

### DISCUSSION THREAD

**Discussion by Exp-life** [sjtug/SJTUThesis#1026] [提问]
在使用Texstudio时突然就无法正确编译参考文献了，问题包括：
  1. 参考文献页直接没有显示
  2. 引用的地方变的跟下图所示的一样
  ![截屏2024-03-25 14 59 39](https://github.com/sjtug/SJTUThesis/assets/157271338/b4f513e1-3e47-447a-8e64-71ece7a0ec8e)
在这之前除了按照Readme中的教程修改了一下Latexmk之外哪里都没动，而且之前原本是可以编译的，但某天打开突然就不行了，请问有大佬遇到过同样的情况知道怎么解决的嘛？

**Comment by LogCreative** [comment]
先尝试 https://github.com/sjtug/SJTUThesis/discussions/1020#discussioncomment-8739430

**Reply by Exp-life** [reply]
好的谢谢 我先试一下

**Reply by Exp-life** [reply]
好像还是不太行orz

**Reply by LogCreative** [reply]
那可能需要提供更详细的日志信息 .log 和 .blg 文件

**Reply by Exp-life** [reply]
哦好的，不过我不太清楚里面的内容orz，就把它都粘贴在这下面了
- **main.log:**
LaTeX Warning: Citation 'sutton2018reinforcement' on page 1 undefined on input 
line 11.


LaTeX Warning: Citation 'mnih2013playing' on page 1 undefined on input line 11.


[1


]

LaTeX Warning: Citation 'e5a1bb8f-41b7-35c6-95cd-8b366d3e99bc' on page 2 undefi
ned on input line 20.

[2]

LaTeX Warning: Citation 'sharpe1964capital' on page 3 undefined on input line 2
2.


LaTeX Warning: Citation 'fama1970efficient' on page 3 undefined on input line 2
4.


LaTeX Warning: Citation 'fama1993common' on page 3 undefined on input line 26.


LaTeX Warning: Citation 'samuelson1970fundamental' on page 3 undefined on input
 line 28.


LaTeX Warning: Citation 'konno1991mean' on page 3 undefined on input line 28.

LaTeX Warning: Empty bibliography on input line 63.

[42] (./contents/acknowledgements.tex)
\tf@thm=\write10
\openout10 = `main.thm'.

[43



] (./main.aux)

LaTeX Warning: There were undefined references.


LaTeX Warning: There were multiply-defined labels.

Package rerunfilecheck Info: File `main.out' has not changed.
(rerunfilecheck)             Checksum: 9561BFD151CFE453D6D1E1F7DAA73C3A;6060.

Package biblatex Warning: Please (re)run Biber on the file:
(biblatex)                main
(biblatex)                and rerun LaTeX afterwards.

Package logreq Info: Writing requests to 'main.run.xml'.
\openout1 = `main.run.xml'.

 ) 
Here is how much of TeX's memory you used:
 49911 strings out of 476179
 1050533 string characters out of 5809543
 1843814 words of memory out of 5000000
 69815 multiletter control sequences out of 15000+600000
 542245 words of font info for 234 fonts, out of 8000000 for 9000
 1348 hyphenation exceptions out of 8191
 103i,27n,116p,1871b,2548s stack positions out of 10000i,1000n,20000p,200000b,200000s

Output written on main.xdv (53 pages, 807508 bytes).

- **main.blg**
This is BibTeX, Version 0.99d (TeX Live 2022)
Capacity: max_strings=200000, hash_size=200000, hash_prime=170003
The top-level auxiliary file: main.aux
I found no \citation commands---while reading file main.aux
I found no \bibdata command---while reading file main.aux
I found no \bibstyle command---while reading file main.aux
You've used 0 entries,
            0 wiz_defined-function locations,
            83 strings with 482 characters,
and the built_in function-call counts, 0 in all, are:
= -- 0
> -- 0
< -- 0
+ -- 0
- -- 0
* -- 0
:= -- 0
add.period$ -- 0
call.type$ -- 0
change.case$ -- 0
chr.to.int$ -- 0
cite$ -- 0
duplicate$ -- 0
empty$ -- 0
format.name$ -- 0
if$ -- 0
int.to.chr$ -- 0
int.to.str$ -- 0
missing$ -- 0
newline$ -- 0
num.names$ -- 0
pop$ -- 0
preamble$ -- 0
purify$ -- 0
quote$ -- 0
skip$ -- 0
stack$ -- 0
substring$ -- 0
swap$ -- 0
text.length$ -- 0
text.prefix$ -- 0
top$ -- 0
type$ -- 0
warning$ -- 0
while$ -- 0
width$ -- 0
write$ -- 0
(There were 3 error messages)

**Reply by LogCreative** [reply]
看不太出来，但是我觉得应该是文档里面有其他的编译错误，如果有其他编译错误的话 biber 大概不能正常工作，可以先试试编译这个仓库里的原始文档看能否编译出来参考文献：如果能够编译出来，可以考虑先删除你论文中的临时文件，然后重新编译（TeXstudio 可能不会自动删除临时文件）；如果原始文档也无法编译出来，说明环境有问题了，建议直接安装 TeX Live 2024。

**Reply by Exp-life** [reply]
嗯好的谢谢 我试一下

**Comment by AlexaraWu** [comment]
看日志实际跑的是 `bibtex` 不是 `biber`，感觉是 TeXstudio 的 bug。

可以试试在主文档 `main.tex` 开头加一条魔术注释，指定文献工具为 `biber`：
```latex
% !TeX TXS-program:bibliography = biber
```

**Reply by Exp-life** [reply]
似乎还是不太行orz

**Comment by Exp-life** [comment]
目前感觉应该是.bbl的问题，第一次编译是没有问题的，但修改后重新编译的话就会出现上述问题，把.bbl文件删除又可以正常编译了。但不知道具体原理是什么orz，.bbl文件也是空白的无法查看
